System/Vector.h

```diff
@@ -66,13 +58,15 @@ namespace RTE {
 		/// Sets the X value of this Vector.
 		/// </summary>
 		/// <param name="newX">A float value that the X value will be set to.</param>
-		void SetX(const float newX) { m_X = newX; }
+		/// <returns>Vector reference to this after the operation.</returns>
+		Vector & SetX(const float newX) { m_X = newX; return *this; }
 
 		/// <summary>
 		/// Sets the X value of this Vector.
 		/// </summary>
 		/// <param name="newX">An int value that the X value will be set to.</param>
-		void SetIntX(const int newX) { m_X = static_cast<float>(newX); }
+		/// <returns>Vector reference to this after the operation.</returns>
+		Vector & SetIntX(const int newX) { m_X = static_cast<float>(newX); return *this; }
```

Comment:
I think we can probably just call this SetX and have it be an overloaded method?

Reply:
Yeah, sounds good to me.